code/lib/components/src/icon/icon.tsx

```diff
@@ -1,39 +1,54 @@
-import React, { ComponentProps, memo } from 'react';
+import React, { FunctionComponent, ComponentProps, memo } from 'react';
 import { styled } from '@storybook/theming';
-import icons, { IconKey } from './icons';
+import { icons, IconKey } from './icons';
 
-import Svg from './svg';
+const Svg = styled.svg`
+  display: inline-block;
+  shape-rendering: inherit;
+  transform: translate3d(0, 0, 0);
+  vertical-align: middle;
 
-const Path = styled.path({
-  fill: 'currentColor',
-});
+  path {
+    fill: currentColor;
+  }
+`;
 
-export interface IconsProps extends ComponentProps<typeof Svg> {
-  icon?: IconKey;
-  symbol?: IconKey;
+/**
+ * An Icon is a piece of visual element, but we must ensure its accessibility while using it.
+ * It can have 2 purposes:
+ *
+ * - *decorative only*: for example, it illustrates a label next to it. We must ensure that it is ignored by screen readers, by setting `aria-hidden` attribute (ex: `<Icon icon="check" aria-hidden />`)
+ * - *non-decorative*: it means that it delivers information. For example, an icon as only child in a button. The meaning can be obvious visually, but it must have a proper text alternative via `aria-label` for screen readers. (ex: `<Icon icon="print" aria-label="Print this document" />`)
+ */
+export interface IconsProps {
+  icon?: IconType;
```

Comment:
Typescript was not happy with me so I added a question mark here. I suspect there is a better way.

Reply:
Why do we need icon and symbol prop? They both pass the same value type but I'm not sure that I'm clear on their purpose.

Reply:
To answer the original question, the question mark is fine!

Reply:
The current interface would allow both `icon` and `symbol` to be `undefined`, which would render an empty `<svg>`. Instead, I believe you want an interface that makes either `icon` or `symbol` required, but not both. The tricky part here is that an interface can't do that! But a type can. I can't easily run this locally at the moment, but try this:

```ts
export type IconsProps = {
  icon: IconType
} | {
  symbol: IconType
}
```

Reply:
@kylegach This is what I see when I try that:

<img width="901" alt="image" src="https://user-images.githubusercontent.com/1123119/182247004-aad560f6-3414-4e9a-89cf-7ca86f5d61ad.png">